app/scripts/views/components/editableRowComponent.js

```diff
@@ -47,20 +51,44 @@
     ' @click="clickSave(data)"><em class="fa fa-save"></em></button><button' +
     ' class="btn btn-default" @click="clickCancel()"><em class="fa' +
     ' fa-times"></em></button></div><div class="buttons" :class="{view:' +
-    ' edit}"><button class="btn btn-info"  @click="clickEdit(data)"><em' +
-    ' class="fa fa-pencil"></em></button><button class="btn btn-danger"' +
+    ' !share}"><div class="input-group"><input id="link-to-share"' +
+    ' v-model="shortenedLink"><button class="btn btn-default btn-share" ' +
+    ' data-clipboard-action="copy" data-clipboard-text={{shortenedLink}}><em' +
+    ' class="fa fa-clipboard" alt="Copy to clipboard"></em></button><button' +
+    ' class="btn btn-default" @click="clickCancel()"><em class="fa' +
+    ' fa-times"></em></button></div></div><div class="buttons"' +
+    ' :class="{view: edit||share}"><button class="btn btn-info"' +
+    ' @click="clickEdit(data)"><em class="fa' +
+    ' fa-pencil"></em></button><button class="btn btn-danger"' +
     ' @click="clickDelete(data)"><em class="fa' +
-    ' fa-trash"></em></button><button class="btn btn-success"><em class="fa' +
-    ' fa-share-alt"></em></button><button class="btn btn-default" @click="clickImport(data)"><em' +
-    ' class="fa fa-filter"></em></button></div></td></tr>',
+    ' fa-trash"></em></button><button class="btn btn-success"' +
+    ' @click="clickShare(data)"><em class="fa' +
+    ' fa-share-alt"></em></button><button class="btn btn-default"' +
+    ' @click="clickImport(data)"><em class="fa' +
+    ' fa-filter"></em></button></div></td></tr>',
     props: [
-      'data','showmodal'
+      'data', 'showmodal'
     ],
     data: function() {
       return {
-        edit: false
+        edit: false,
+        share: false,
+        shortenedLink: '---'
       };
     },
+    ready: function() {
```

Comment:
Need to debug a little bit. This function will be called at least three times in one click.